voir/instruments/gpu/cuda.py

```diff
@@ -0,0 +1,75 @@
+IMPORT_ERROR = None
+try:
+    from pynvml import nvmlInit
+    from pynvml.smi import nvidia_smi
+except ImportError as err:
+    IMPORT_ERROR = err
+
+
+def fix_num(n):
+    if n == "N/A":
+        n = None
+    return n
+
+
+def parse_gpu(gpu, gid):
+    mem = gpu["fb_memory_usage"]
+    used = fix_num(mem["used"])
+    total = fix_num(mem["total"])
+    compute = fix_num(gpu["utilization"]["gpu_util"])
+    if compute:
+        compute /= 100
+    return {
+        "device": gid,
+        "product": gpu["product_name"],
+        "memory": {
+            "used": used,
+            "total": total,
+        },
+        "utilization": {
+            "compute": compute,
+            "memory": total and used and (used / total),
+        },
+        "temperature": fix_num(gpu["temperature"]["gpu_temp"]),
+        "power": fix_num(gpu["power_readings"]["power_draw"]),
+        "selection_variable": "CUDA_VISIBLE_DEVICES",
+    }
+
+
+def is_available():
+    return IMPORT_ERROR is None
```

Comment:
I'm not sure that's right. Does the library raise an ImportError if there are no CUDA devices? In the original gpu.py we also catch NVMLError_DriverNotLoaded and NVMLError_LibraryNotFound during nvml_init.

Reply:
Yes, this only needs for the library to be installed but when selecting the backend we try to instantiate the DeviceSMI object this is when the `NVMLError_DriverNotLoaded ` will get raised. 
The backend will ignore the error if we are in deduce mode, 
but the error will be raised to the user if he selects cuda explicitly.

```
select_backend()         <= does not raise anything, if nothing is available; it will assume CPU
select_backend(cuda) <= raise `NVMLError_DriverNotLoaded ` / `NVMLError_LibraryNotFound ` to the user
```